src/KafkaClient/Protocol/KafkaEncoder.cs

```diff
@@ -230,7 +230,7 @@ private static IResponse MetadataResponse(IRequestContext context, ArraySegment<
                     var host = reader.ReadString();
                     var port = reader.ReadInt32();
                     string rack = null;
-                    if (context.ApiVersion >= 1) {
+                    if (context.ApiVersion >= 2) {
                         rack = reader.ReadString();
```

Comment:
rack is part of version 1:
Metadata Response (Version: 1) => [brokers] controller_id [topic_metadata] 
  brokers => node_id host port rack 
    node_id => INT32
    host => STRING
    port => INT32
    rack => NULLABLE_STRING
...